dev/devicelab/bin/run.dart

```diff
@@ -35,6 +36,16 @@ bool exitOnFirstTestFailure;
 /// The device-id to run test on.
 String deviceId;
 
+/// File containing a service account token.
+///
+/// If passed, the test run results will be recorded by Flutter infrastructure.
+String serviceAccountFile;
+
+/// Key for the task to upload to in Flutter infrastructure.
+/// 
+/// Only used if [serviceAccountFile] is passed.
+String taskKey;
```

Comment:
I think we have all the resources needed to compute the task key here, but then we make it harder to change it on the backend.

Reply:
Is it possible to only use the commit?

Reply:
We can pass the commit instead, and update cocoon to take the commit sha + task name.

Reply:
Can we assume git will be on the host running this test? If so, we could just pull this from git

Reply:
I started to work on the backend change for this, and realized we're going to need to send the branch as well.

Do you know if recipes knows about the branch?

Reply:
They now about the release ref dev, stable, etc. But I assume getting the branch is possible using git.

Reply:
I got a bit further in the backend, and we can scratch pulling the branch here. I removed the git commit sha flag, and pull it in the cocoon client via git.

If we switch to building these tests as a binary for the bot, we'll need to pass the commit sha as a flag.